### PR TITLE
[FIX] website: optmize the asset disabling function

### PR DESCRIPTION
**Pull request by ramiadavid**
Before this commit, the behavior disabling the unused snippet assets performed two checks:

1. He was looking for its use in the snippet template
2. He was looking for its use in the HTML fields
3. Checked on the occurrences of step 1 and 2 and return result.

In many cases there are already coincidences in the first step, making the second step unnecessary since this second one is the slowest. Matches are now checked between steps 1 and 2 to skip the second if matches are already found.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website/models/website.py [hunk 1]
```diff
@@ -1437,6 +1437,9 @@ def _is_snippet_used(self, snippet_module, snippet_id, asset_version, asset_type
             match = re.search('<([^>]*class="[^>]*)>', snippet_template_html)
             snippet_occurences.append(match.group())
 
+        if self._check_snippet_used(snippet_occurences, asset_type, asset_version):
+            return True
```
Comment (qsm-odoo) [inline addons/website/models/website.py:1441]: Indeed, it seems like a good thing to do. Although, if I am not mistaken, you are re-calling `_check_snippet_used` with the already checked "snippet occurrences" the second time you call it. Maybe you can improve this? :+1:

File: addons/website/models/website.py [hunk 2]
```diff
@@ -1449,6 +1452,12 @@ def _is_snippet_used(self, snippet_module, snippet_id, asset_version, asset_type
         for r in results:
             snippet_occurences.append(r[0][0])
 
+        if self._check_snippet_used(snippet_occurences, asset_type, asset_version):
+            return True
+
+        return False
```
Comment (qsm-odoo) [inline addons/website/models/website.py:1458]:
```suggestion
        return self._check_snippet_used(snippet_occurences, asset_type, asset_version)
```

File: addons/website/models/website.py [hunk 3]
```diff
@@ -1437,6 +1437,9 @@ def _is_snippet_used(self, snippet_module, snippet_id, asset_version, asset_type
             match = re.search('<([^>]*class="[^>]*)>', snippet_template_html)
             snippet_occurences.append(match.group())
 
+        if self._check_snippet_used(snippet_occurences, asset_type, asset_version):
+            return True
+
         # As well as every snippet dropped in html fields
         self.env.cr.execute(sql.SQL(" UNION ").join(
             sql.SQL("SELECT regexp_matches({}, {}, 'g') FROM {}").format(
```
Comment (qsm-odoo) [inline addons/website/models/website.py:1445]: Now that I check this code, I wonder if there is not another more important thing to improve here: if I am not mistaken we are performing this same SQL search for each snippet... for each version of each asset type and only check if this is the version of asset type we are looking for afterwards? We should make this search only once per snippet and check all versions and asset types afterwards.

We'll handle that part ourselves of course, unless you want to @ramiadavid

**Review by qsm-odoo — CHANGES_REQUESTED** [review]
Thank you for the suggestion :+1: 

I am asking for two changes if you don't mind + pointing at a potentially more important one we can handle ourselves.

I like your commit message but could you also adapt it to fix some typos and respect our conventions?
- optmize -> optimize
- He was looking for its use -> It was looking for occurrences
- 72 characters per line max

Thank you :+1:

File: addons/website/models/website.py (hunk 1 above)
Comment (ramiadavid) [reply addons/website/models/website.py:1441]: done

File: addons/website/models/website.py (hunk 2 above)
Comment (ramiadavid) [reply addons/website/models/website.py]: done

File: addons/website/models/website.py (hunk 3 above)
Comment (ramiadavid) [reply addons/website/models/website.py:1445]: @qsm-odoo This SQL query is a bit complex for my SQL knowledge, if you can do it, better

Comment (qsm-odoo) [reply addons/website/models/website.py:1445]: Sure, we'll do that, in master probably :+1:

**Review by qsm-odoo — APPROVED** [review]
Nice, thank you for the quick and good adaptation :+1: 

As discussed we'll review to further improve this on our side :+1: 

@robodoo r+

**Pull request closed without merging**